src/azul/service/manifest_service.py

```diff
@@ -597,6 +600,35 @@ def create_file(self) -> Tuple[str, Optional[str]]:
         raise NotImplementedError
 
 
+class BulkDownloadManifestGenerator(StreamingManifestGenerator):
+
+    @property
+    def content_type(self) -> str:
+        return 'text/plain'
+
+    @property
+    def file_name_extension(self):
+        return 'txt'
+
+    @property
+    def entity_type(self) -> str:
+        return 'files'
+
+    def write_to(self, output: IO[str]) -> Optional[str]:
+        output.write('--create-dirs\n\n')
+        output.write('--compressed\n\n')
+        for hit in self._create_request().scan():
+            doc = self._hit_to_doc(hit)
+            assert isinstance(doc, dict)
```

Comment:
Why is this necessary? The next line uses `[]` so it would fail if doc were not a dict.